lib/cdo/aws/cloudfront.rb

```diff
@@ -171,19 +171,28 @@ def self.distribution_config(app, origin, aliases, ssl_cert=nil)
           Prefix: cloudfront[:log][:prefix]
         },
         Origins: [
-          {
-            Id: 'cdo',
-            CustomOriginConfig: {
-              OriginProtocolPolicy: 'match-viewer',
-              OriginSSLProtocols: %w(TLSv1.2 TLSv1.1)
-            },
-            DomainName: origin,
-            OriginPath: '',
-            OriginShield: {
-              Enabled: true,
-              OriginShieldRegion: {Ref: 'AWS::Region'}
+          *HTTP_CACHE.keys.map do |app_name|
+            proxy = (app == :hourofcode) ? :pegasus : app
+            {
+              Id: app_name == proxy ? 'cdo' : app_name,
+              CustomOriginConfig: {
+                OriginProtocolPolicy: 'match-viewer',
+                OriginSSLProtocols: %w(TLSv1.2 TLSv1.1)
+              },
+              DomainName: origin,
+              OriginPath: '',
+              OriginShield: {
+                Enabled: true,
+                OriginShieldRegion: {Ref: 'AWS::Region'}
+              },
+              OriginCustomHeaders: app_name == proxy ? [] : [
+                {
+                  HeaderName: 'X-Cdo-Backend',
+                  HeaderValue: app_name
+                }
+              ]
             }
-          },
+          end,
```

Comment:
This is creating an extra 'Origin' in addition to `'cdo'`, which the distribution can use as a `proxy` target (so the pegasus/hourofcode distributions will have an extra `dashboard` origin, and dashboard will have an extra `pegasus` origin). The only difference in the second origin is that it sets a custom HTTP request header `X-Cdo-Backend` to `pegasus`/`dashboard`, which nginx then uses as an override when selecting the proper backend app.